src/Microsoft.OData.Core/UriParser/SemanticAst/ExpandedCountSelectItem.cs

```diff
@@ -0,0 +1,53 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ExpandedCountSelectItem.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    using Aggregation;
+    using Microsoft.OData.Edm;
+
+    /// <summary>
+    /// This represents one level of expansion for a particular expansion tree.
+    /// </summary>
+    public sealed class ExpandedCountSelectItem : ExpandedReferenceSelectItem
+    {
+        /// <summary>
+        /// Create an Expand item using a nav prop, its entity set and a SelectExpandClause
+        /// </summary>
+        /// <param name="pathToNavigationProperty">the path to the navigation property for this expand item, including any type segments</param>
+        /// <param name="navigationSource">the navigation source for this ExpandItem</param>
+        /// <param name="filterOption">A filter clause for this expand (can be null)</param>
+        /// <param name="searchOption">A search clause for this expand (can be null)</param>
+        /// <exception cref="System.ArgumentNullException">Throws if input pathToNavigationProperty is null.</exception>
+        public ExpandedCountSelectItem(ODataExpandPath pathToNavigationProperty, IEdmNavigationSource navigationSource, FilterClause filterOption, SearchClause searchOption)

```

Comment:
I think it would be helpful if you could provide an concrete example of a path that this class represents, and how the parts of the path that the different parameters represent.